var/spack/repos/builtin/packages/boost-mpl-cartesian_product/package.py

```diff
@@ -0,0 +1,40 @@
+##############################################################################
```

Comment:
Package names cannot contain underscores. `spack create` should handle this conversion for you when you create the package. This is what's causing the unit tests to fail.